src/coreclr/jit/valuenum.cpp

```diff
@@ -6798,6 +6798,7 @@ static genTreeOps genTreeOpsIllegalAsVNFunc[] = {GT_IND, // When we do heap memo
                                                  GT_OBJ,      // May reference heap memory.
                                                  GT_BLK,      // May reference heap memory.
                                                  GT_INIT_VAL, // Not strictly a pass-through.
+                                                 GT_BITCAST,  // Elide sequential bitcasts
```

Comment:
```suggestion
                                                 GT_BITCAST,  // Needs to encode the target type.
```
The reason we need the special handling is because `BITCAST<long>(x) != BITCAST<double>(x)`, and the "default" `VNFunc(GT_BITCAST)` VN func would only have one operand (`x`).